### PR TITLE
feat: Add create template from the UI

### PR DESCRIPTION
**Pull request by BrunoQuaresma**

### CODE REVIEW

File: site/src/xServices/createTemplate/createTemplateXService.ts [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+              invoke: {
+                src: "waitForJobToBeCompleted",
+                onDone: [
+                  {
+                    target: "loadingMissingParameters",
+                    cond: "hasMissingParameters",
+                    actions: ["assignVersion"],
+                  },
+                  {
+                    target: "#createTemplate.idle",
```
Comment (Kira-Pilot) [inline site/src/xServices/createTemplate/createTemplateXService.ts:172]: This `#createTemplate.idle` goes back up to line 115, right?

File: site/src/xServices/createTemplate/createTemplateXService.ts [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+            },
+            created: {
+              type: "final",
+            },
+          },
+        },
+      },
+    },
+    {
+      services: {
```
Comment (Kira-Pilot) [inline site/src/xServices/createTemplate/createTemplateXService.ts:243]: I would probably break these services into a separate file just to minimize scrolling. I find myself scrolling a lot when trying to understand XState machines :)

File: site/src/xServices/createTemplate/createTemplateXService.ts (hunk 1 above)
Comment (BrunoQuaresma) [reply site/src/xServices/createTemplate/createTemplateXService.ts:172]: Yes

File: site/src/xServices/createTemplate/createTemplateXService.ts (hunk 2 above)
Comment (BrunoQuaresma) [reply site/src/xServices/createTemplate/createTemplateXService.ts:243]: Yes. Unfortunately XState typegen does not work pretty well with that but I think I can create separated services tho. Let's see if it gets better.

File: site/src/i18n/en/createTemplatePage.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+      "title": "General info",
+      "description": "The name is used to identify the template on the URL and also API. It has to be unique across your organization."
+    },
+    "displayInfo": {
+      "title": "Display info",
+      "description": "Set the name that you want to use to display your template, a helpful description and icon."
+    },
+    "schedule": {
+      "title": "Schedule",
+      "description": "Define when a workspace create from this template is going to stop."
```
Comment (presleyp) [inline site/src/i18n/en/createTemplatePage.json:14]:
```suggestion
      "description": "Define when a workspace created from this template is going to stop."
```

File: site/src/i18n/en/createTemplatePage.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    "operations": {
+      "title": "Operations",
+      "description": "Allow or not users to run specific actions on the workspace."
```
Comment (presleyp) [inline site/src/i18n/en/createTemplatePage.json:18]:
```suggestion
      "description": "Allow or disallow users to run specific actions on the workspace."
```

File: site/src/i18n/en/createTemplatePage.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    "parameters": {
+      "title": "Template params",
+      "description": "Those params are provided by your template's Terraform configuration."
```
Comment (presleyp) [inline site/src/i18n/en/createTemplatePage.json:22]:
```suggestion
      "description": "These params are provided by your template's Terraform configuration."
```

**Comment by presleyp** [comment]
Doing some QA:
- the starter template and create template pages look nice!
- creating from an AWS starter template failed, but I think that's expected. It showed logs, and Retry and Cancel worked.
- creating from a Docker starter template succeeded
- both pages correctly show a validation error if I leave out the name
- emoji picker works
- filters on starter templates page work

- if I focus the Schedule input and then scroll the page, the number changes. That might be a feature of a number input but it surprised me.
- The display name and description inputs don't allow spaces
- I tried to create a template without a file and saw "Error" under the template name field - I'm guessing it was in response to the lack of a file but it wasn't informative and looked like it was complaining about the name (which was filled in)

**Comment by presleyp** [comment]
I still had an error while provisioning my own template (probably user error) but the issues I found are fixed!

**Review by presleyp — APPROVED** [review]

**Pull request merged**